src/debugAdapter/runtimeInterface.ts

```diff
@@ -295,4 +280,32 @@ export default class RuntimeInterface extends EventEmitter {
       throw new Error('Debug session is undefined');
     }
   }
+
+  /**
+   * Validate if the network parameter was set. Case the network is empty,
+   * a temp network is created to generate a provider url
+   *
+   * @param config Represents the truffle configuration file.
+   * @param args Represents the arguments needed to initiate a new Truffle `DebugSession` request.
+   * @returns
+   */
+  private validateNetwork(config: Config, args: Required<DebuggerTypes.DebugArgs>): void {
+    if (args.providerUrl) {
+      // Adds a temporary network to get the provider url
+      config.network = 'temp_network';
+      config.networks.temp_network = {
+        url: args.providerUrl,
+        network_id: '*',
+      };
+      return;
+    }
```

Comment:
We are just creating a dummy network just to comply with `Environment.detect`. Isn't better if we embed this logic directly into the `attach` method when setting `options: DebuggerOptions`?

Reply:
Hey @acuarica,

I think I did not understand your point. You are suggesting vanish out the `validateNetwork(config, args)` method and put the logic right before the `await Environment.detect(config);` calling, is that?